wagtail/admin/checks.py

```diff
@@ -188,8 +188,17 @@ def wagtail_admin_base_url_check(app_configs, **kwargs):
 @register("file_overwrite")
 def file_overwrite_check(app_configs, **kwargs):
     from django.conf import settings
-
-    file_storage = getattr(settings, "DEFAULT_FILE_STORAGE", None)
+    from django import VERSION as DJANGO_VERSION
+
+    if DJANGO_VERSION >= (4, 2):
+        try:
+            file_storage = getattr(settings, "STORAGES")["default"]["BACKEND"]
+        except AttributeError:
+            file_storage = getattr(settings, "DEFAULT_FILE_STORAGE", None)
+    elif DJANGO_VERSION >= (5, 1):
```

Comment:
This `elif` doesn't make sense here - if DJANGO_VERSION is greater than (5,1) then the `if DJANGO_VERSION >= (4, 2):` will have been taken, so this code is never reached.